// @material-ui/lab/node/internal/pickers/hooks/useKeyDown.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useGlobalKeyDown = useGlobalKeyDown;
exports.keycode = void 0;

var React = _interopRequireWildcard(require("react"));

var _utils = require("@material-ui/utils");

function runKeyHandler(event, keyHandlers) {
  // tslint:disable-next-line deprecation IE11
  const handler = keyHandlers[event.keyCode];

  if (handler) {
    handler(); // if event was handled prevent other side effects (e.g. page scroll)

    event.preventDefault();
  }
}

function useGlobalKeyDown(active, keyHandlers) {
  const keyHandlersRef = React.useRef(keyHandlers);
  keyHandlersRef.current = keyHandlers;
  (0, _utils.unstable_useEnhancedEffect)(() => {
    if (active) {
      const handleKeyDown = nativeEvent => {
        runKeyHandler(nativeEvent, keyHandlersRef.current);
      };

      document.addEventListener('keydown', handleKeyDown);
      return () => {
        document.removeEventListener('keydown', handleKeyDown);
      };
    }

    return undefined;
  }, [active]);
}

const keycode = {
  ArrowUp: 38,
  ArrowDown: 40,
  ArrowLeft: 37,
  ArrowRight: 39,
  Enter: 13,
  Home: 36,
  End: 35,
  PageUp: 33,
  PageDown: 34,
  Esc: 27
};
exports.keycode = keycode;
