// @material-ui/lab/node/LoadingButton/LoadingButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.styles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var React = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _clsx = _interopRequireDefault(require("clsx"));

var _utils = require("@material-ui/utils");

var _utils2 = require("@material-ui/core/utils");

var _styles = require("@material-ui/core/styles");

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

var _CircularProgress = _interopRequireDefault(require("@material-ui/core/CircularProgress"));

var _jsxRuntime = require("react/jsx-runtime");

const styles = () => ({
  /* Styles applied to the root element. */
  root: {},

  /* Styles applied to the root element if `loading={true}`. */
  loading: {},

  /* Styles applied to the loadingIndicator element. */
  loadingIndicator: {
    position: 'absolute',
    visibility: 'visible',
    display: 'flex'
  },

  /* Styles applied to the loadingIndicator element if `loadingPosition="center"`. */
  loadingIndicatorCenter: {
    left: '50%',
    transform: 'translate(-50%)'
  },

  /* Styles applied to the loadingIndicator element if `loadingPosition="start"`. */
  loadingIndicatorStart: {
    left: 14
  },

  /* Styles applied to the loadingIndicator element if `loadingPosition="end"`. */
  loadingIndicatorEnd: {
    right: 14
  },

  /* Styles applied to the endIcon element if `loading={true}` and `loadingPosition="end"`. */
  endIconLoadingEnd: {
    visibility: 'hidden'
  },

  /* Styles applied to the startIcon element if `loading={true}` and `loadingPosition="start"`. */
  startIconLoadingStart: {
    visibility: 'hidden'
  },

  /* Styles applied to the label element if `loading={true}` and `loadingPosition="center"`. */
  labelLoadingCenter: {
    visibility: 'hidden'
  }
});

exports.styles = styles;
const LoadingIndicator = /*#__PURE__*/(0, _jsxRuntime.jsx)(_CircularProgress.default, {
  color: "inherit",
  size: 16
});
const LoadingButton = /*#__PURE__*/React.forwardRef(function LoadingButton(props, ref) {
  const {
    children,
    classes,
    className,
    disabled = false,
    loading = false,
    loadingIndicator = LoadingIndicator,
    loadingPosition = 'center'
  } = props,
        other = (0, _objectWithoutPropertiesLoose2.default)(props, ["children", "classes", "className", "disabled", "loading", "loadingIndicator", "loadingPosition"]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_Button.default, (0, _extends2.default)({
    className: (0, _clsx.default)(classes.root, className, loading && classes.loading),
    disabled: disabled || loading,
    ref: ref,
    classes: {
      startIcon: classes[`startIcon${loading ? 'Loading' : ''}${(0, _utils2.capitalize)(loadingPosition)}`],
      endIcon: classes[`endIcon${loading ? 'Loading' : ''}${(0, _utils2.capitalize)(loadingPosition)}`],
      label: classes[`label${loading ? 'Loading' : ''}${(0, _utils2.capitalize)(loadingPosition)}`]
    }
  }, other, {
    children: [loading && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: (0, _clsx.default)(classes.loadingIndicator, classes[`loadingIndicator${(0, _utils2.capitalize)(loadingPosition)}`]),
      children: loadingIndicator
    }), children]
  }));
});
process.env.NODE_ENV !== "production" ? LoadingButton.propTypes
/* remove-proptypes */
= {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // |     To update them edit the d.ts file and run "yarn proptypes"     |
  // ----------------------------------------------------------------------

  /**
   * The content of the component.
   */
  children: _propTypes.default.node,

  /**
   * Override or extend the styles applied to the component.
   */
  classes: _propTypes.default.object,

  /**
   * @ignore
   */
  className: _propTypes.default.string,

  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: _propTypes.default.bool,

  /**
   * If `true`, the loading indicator is shown.
   * @default false
   */
  loading: _propTypes.default.bool,

  /**
   * Element placed before the children if the button is in loading state.
   * @default <CircularProgress color="inherit" size={16} />
   */
  loadingIndicator: _propTypes.default.node,

  /**
   * The loading indicator can be positioned on the start, end, or the center of the button.
   * @default 'center'
   */
  loadingPosition: (0, _utils.chainPropTypes)(_propTypes.default.oneOf(['start', 'end', 'center']), props => {
    if (props.loadingPosition === 'start' && !props.startIcon) {
      return new Error(`Material-UI: The loadingPosition="start" should be used in combination with startIcon.`);
    }

    if (props.loadingPosition === 'end' && !props.endIcon) {
      return new Error(`Material-UI: The loadingPosition="end" should be used in combination with endIcon.`);
    }

    return null;
  })
} : void 0;

var _default = (0, _styles.withStyles)(styles, {
  name: 'MuiLoadingButton'
})(LoadingButton);

exports.default = _default;
